src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -653,6 +702,30 @@ private void PopulateProtocol()
             }
         }
 
+        public static string GetLocalDBInstance(string dataSource, out bool error)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(dataSource), "Empty data source");
+
+            error = false;
+
+            string instanceName = null;
+
+            string workingDataSource = dataSource.ToLower();
```

Comment:
Note that `ToLower()` does a culture-sensitive lower case based on the current culture. Should this be `ToLowerInvariant()`?